src/Terrain/MapMaster.java

```diff
@@ -237,6 +238,29 @@ public boolean isLocationEmpty(Unit unit, int x, int y)
     return empty;
   }
 
+  public void setOwner(Commander owner, XYCoord xyc)
+  {
+    setOwner(owner, xyc.x, xyc.y);
+  }
+  public void setOwner(Commander owner, int x, int y)
+  {
+    MapLocation loc = getLocation(x, y);
+    Commander prevOwner = loc.getOwner();
+    if( prevOwner == owner )
+      return;
+
+    XYCoord xyc = loc.getCoordinates();
```

Comment:
Seems a bit wonky to have the (x, y) overload be the "base" one when it just reconstitutes the `XYCoord` halfway through. Not that this is like, important.

Reply:
`loc.getCoordinates()` doesn't create a coord, so it's one less total allocation.
Truly, the pinnacle of optimizations.